pkg/internal/transform/routes.go

```diff
@@ -20,8 +20,10 @@ const (
 	UnmatchPath = UnmatchType("path")
 	// UnmatchWildcard sets the route field to a generic asterisk symbol
 	UnmatchWildcard = UnmatchType("wildcard")
+	// UnmatchAuto detects the route field using a heuristic
+	UnmatchAuto = UnmatchType("auto")
 
-	UnmatchDefault = UnmatchWildcard
+	UnmatchDefault = UnmatchAuto
```

Comment:
I'm not sure if we should default to the slowest option. Please check the benchmark results in this PR: https://github.com/grcevski/ebpf-autoinstrument/pull/2

Reply:
I worked on removing bunch of allocations and remembering good words so we get O(1) check when the routes are valid. I got it to run faster and it's not far off as much as it used to:

```
cpu: 12th Gen Intel(R) Core(TM) i7-1280P
BenchmarkRoutesProvider_Wildcard-20     	 1625084	       743.3 ns/op	     272 B/op	       5 allocs/op
BenchmarkRoutesProvider_Heuristic-20    	  909910	      1128 ns/op	     336 B/op	      12 allocs/op
PASS
ok  	github.com/grafana/beyla/pkg/internal/transform	3.009s

```

I think we need to check how much is this from the overall overhead, it might be worth it for getting better first user experience. WDYT?

Reply:
Yeah, I agree.